src/couch/src/couch_server.erl

```diff
@@ -525,7 +525,7 @@ handle_call({delete, DbName, Options}, _From, Server) ->
         DelOpt = [{context, delete} | Options],
 
         % Make sure and remove all compaction data
-        delete_compaction_files(DbNameList, DelOpt),
+        delete_compaction_files(DbNameList, Options),
```

Comment:
I don't think it is sufficient. The `delete` context is added back [here](https://github.com/apache/couchdb/blob/master/src/couch/src/couch_bt_engine.erl#L768)

Reply:
It's not added back, local `delete_compaction_files` calls `couch_db_engine:delete_compaction_files/4` not `couch_db_engine:delete_compaction_files/1`, but it's a good catch, though I haven't seen it in action, I suspect it's a dead (i.e.) never called function.

Reply:
@iilyak nope, not dead, just quite specific, needs recreate or invalid header on main db file to be triggered. I've pushed an update to address that. Good catch!